src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.Comparison.cs

```diff
@@ -0,0 +1,412 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Text;
+using System.Text.Unicode;
+using System.Numerics;
+
+#pragma warning disable SA1121 // Use built-in type alias
+using SkipChecks = System.Boolean;
+using CheckBytes = System.Byte;
+using CheckChars = System.Char;
+using CheckNonAscii = System.Byte;
+
+namespace System.Buffers.Text
+{
+    public static partial class Ascii
+    {
+        public static bool TryGetHashCode(ReadOnlySpan<byte> value, out int hashCode)
```

Comment:
What's the use case for these case-sensitive APIs?  Why would you want to hash bytes but only if they're ASCII, especially when you can just call IsAscii yourself?

You can hash any `ReadOnlySpan<byte>` with:
```C#
HashCode hc = default;
hc.AddBytes(value);
... = hc.ToHashCode();
```
and if it was really valuable to be able to do that with less ceremony, we could add one-shot static methods to let you do:
```C#
... = HashCode.Combine(spanOfBytes);
```
at which point you can do the ASCII check yourself, e.g.
```C#
if (Ascii.IsAscii(value)) { Use(HashCode.Combine(spanOfBytes)); }
```